runconfig/hostconfig_unix.go

```diff
@@ -30,9 +30,12 @@ func validateNetMode(c *container.Config, hc *container.HostConfig) error {
 	if err != nil {
 		return err
 	}
-	if hc.UTSMode.IsHost() && c.Hostname != "" {
+	if (hc.UTSMode.IsHost() || hc.UTSMode.IsContainer()) && c.Hostname != "" {
 		return ErrConflictUTSHostname
 	}
+	if hc.UTSMode.IsContainer() && c.Domainname != "" {
```

Comment:
I know it didn't even include a check for this at all previously (:heart:!!!!), but this one should include `IsHost` as well, right?

Reply:
Yes, I did actually have the `IsHost` in there, but I removed it as there was no test for a conflict between `IsHost` and `Domainname`, so I thought perhaps it was a valid combination.

Do you want me to add back the check?

Reply:
I can also add some tests for `IsHost` with `Domainname`/`Hostname`.

Reply:
@tianon I have added the error and added two tests to check for this.